scipy/linalg/_decomp_schur.py

```diff
@@ -100,9 +100,9 @@ def schur(a, output='real', lwork=None, overwrite_a=False, sort=None,
 
     >>> T2, Z2 = schur(A, output='complex')
     >>> T2
-    array([[ 2.65896708, -1.22839825+1.32378589j,  0.42590089+1.51937378j],
-           [ 0.        , -0.32948354+0.80225456j, -0.59877807+0.56192146j],
-           [ 0.        ,  0.                    , -0.32948354-0.80225456j]])
+    array([[ 2.65896708, -1.22839825+1.32378589j,  0.42590089+1.51937378j], # may vary
+           [ 0.        , -0.32948354+0.80225456j, -0.59877807+0.56192146j], # may vary
+           [ 0.        ,  0.                    , -0.32948354-0.80225456j]]) # may vary
```

Comment:
Strictly speaking, a single `# may vary` anywhere in the output is enough, cf https://github.com/scipy/scipy/blob/main/tools/refguide_check.py#L578
(here `want` is lines 103-105, `array([[...]])`)

So to me at least this reads like you're really stressing that each row is essentially random, not just the whole matrix (?) --- is this your intention?

Reply:
I didn't know that, but I did see some fairly striking differences:

```
File "build-install/lib/python3.11/site-packages/scipy/linalg/_decomp_schur.py", line 102, in schur
Failed example:
    T2
Expected:
    array([[ 2.65896708, -1.22839825+1.32378589j,  0.42590089+1.51937378j],
           [ 0.        , -0.32948354+0.80225456j, -0.59877807+0.56192146j],
           [ 0.        ,  0.                    , -0.32948354-0.80225456j]])
Got:
    array([[ 2.65896708-3.33066907e-16j, -1.58397876+8.67399929e-01j,
             0.42590083+1.51937380e+00j],
           [ 0.        +0.00000000e+00j, -0.32948354+8.02254558e-01j,
            -0.39035561+7.22435738e-01j],
           [ 0.        +0.00000000e+00j,  0.        +0.00000000e+00j,
            -0.32948354-8.02254558e-01j]])
```

Reply:
Could you double-check that the result on your system is a correct Schur decomp? Basically, that indeed `A = Z T Z^H` modulo floating-point noise? 

My suspicion is that the root of the problem is that the Schur decomp is not unique, and that the example and your system just return two different, perfectly legal, factorizations. If that's indeed the case, a single `# may vary` incantation anywhere in the `want` text would be enough to paper over the hard failure.  (or `# random` -- this is functionally equivalent but maybe not very appropriate for readability here? --- both are essentially a `# doctest: +SKIP`, only apply to `want` not `source` in the doctest parlance) . 

A long-term fix would be to adjust the example to either do not list individual matrices or add some text (maybe also a reference?) that indeed there is variability and that's OK. But that's probably a separate doc enhancement anyway.

Reply:
I pushed in the simplification to use a single `# may vary`. If you want me to check the linalg thing I may need to be spoon fed how to do that, unfortunately. I suppose I might hope that the full test suite wouldn't pass if I had a fundamental local issue.

I'm happy for Lucas to push in the refguide shims he needs on his local machine as well, if you want.